.github/workflows/unit-test.yml

```diff
@@ -0,0 +1,25 @@
+name: Build and run tests
+
+on:
+  push:
+    branches: [ $default-branch ]
+  pull_request:
+    branches: [ $default-branch ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up JDK 11
```

Comment:
we should probably go with JDK 8 to maintain compatibility with other projects

Reply:
Ah, yeah. Good catch!

I'm still testing this, btw. I was hoping to see it run here, but testing these things is kind of a pain.